src/bigwig/BigWigReader.ts

```diff
@@ -237,9 +243,9 @@ export class BigWigReader {
      * @param treeOffset Location of the R+ tree that stores the data we're interested.
      * @param decodeFunction 
      */
-    private async loadData<T>(startChrom: string, startBase: number, endChrom: string, endBase: number, 
-                treeOffset: number, streamMode: boolean, decodeFunction: DecodeFunction<T>, 
-                loadFunction: LoadFunction<T>): Promise<void> {
+    private async loadData<T>(startChrom: string, startBase: number, endChrom: string, endBase: number,
+        treeOffset: number, streamMode: boolean, decodeFunction: DecodeFunction<T>,
+        loadFunction: LoadFunction<T>, restParser?: ParseFunction<T>): Promise<void> {
```

Comment:
Oh, I forgot that this gets called for both bigBeds and bigWigs. An alternative to make this *not* change the function signature is to make `decodeBedData` actually be a function with the `restParser` argument that returns the actual decode function.

Then, in `readBigBedData` you pass `decodeBedData(restParser || parseBigBed as any)` for the decode function.

Reply:
@jackh726 This is tricky. ` decodeBedData(restParser || parseBigBed as any)` can be like a closure- a function that returns a function. But I am not sure how to get the other variables that the decode function need (like filterStartChromIndex) in this approach as only the `restParser || parseBigBed as any` is available initial setup in `readBigBedData`. It also gets tricky in that `decodeZoomData` does not used restParser.

Reply:
I used an interface as a function seemed verbose. Let me know if you prefer a function.